### ISSUE TITLE
[Bug] REI crashes when attempting to render the Estrogen Patch sequenced assembly recipe

### ISSUE THREAD

**Issue by thrilliams** [MayaqqDev/Estrogen#100]
Minecraft 1.20.1
Create Fabric 0.5.1-f-build.1417+mc1.20.1
Create: Estrogen 4.2.7+1.20.1-fabric
REI 12.1.725+fabric

trace as follows:

```
java.lang.IndexOutOfBoundsException: index (1) must be less than size (1)
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1355)
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:1337)
	at com.google.common.collect.SingletonImmutableList.get(SingletonImmutableList.java:44)
	at com.simibubi.create.compat.rei.category.sequencedAssembly.ReiSequencedAssemblySubCategory$AssemblySpouting.addFluidIngredients(ReiSequencedAssemblySubCategory.java:88)
	at com.simibubi.create.compat.rei.category.SequencedAssemblyCategory.addWidgets(SequencedAssemblyCategory.java:86)
	at com.simibubi.create.compat.rei.category.CreateRecipeCategory.setupDisplay(CreateRecipeCategory.java:284)
	at com.simibubi.create.compat.rei.category.CreateRecipeCategory.setupDisplay(CreateRecipeCategory.java:46)
	at me.shedaniel.rei.impl.client.gui.widget.DisplayTooltipComponent.<init>(DisplayTooltipComponent.java:61)
	at me.shedaniel.rei.plugin.client.runtime.DefaultClientRuntimePlugin$DisplayFavoriteEntry.lambda$new$1(DefaultClientRuntimePlugin.java:340)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183)
	at me.shedaniel.rei.plugin.client.runtime.DefaultClientRuntimePlugin$DisplayFavoriteEntry$1.getTooltip(DefaultClientRuntimePlugin.java:385)
	at me.shedaniel.rei.impl.client.entry.type.types.RenderingEntryDefinition$DeferredRenderer.getTooltip(RenderingEntryDefinition.java:88)
	at me.shedaniel.rei.impl.common.entry.AbstractEntryStack.getTooltip(AbstractEntryStack.java:195)
	at me.shedaniel.rei.api.common.entry.EntryStack.getTooltip(EntryStack.java:226)
	at me.shedaniel.rei.impl.client.gui.widget.EntryWidget.getCurrentTooltip(EntryWidget.java:453)
	at me.shedaniel.rei.impl.client.gui.widget.EntryWidget.queueTooltip(EntryWidget.java:441)
	at me.shedaniel.rei.impl.client.gui.widget.DisplayedEntryWidget.queueTooltip(DisplayedEntryWidget.java:60)
	at me.shedaniel.rei.impl.client.gui.widget.EntryWidget.method_25394(EntryWidget.java:391)
	at me.shedaniel.rei.impl.client.gui.widget.region.RegionEntryWidget.method_25394(RegionEntryWidget.java:84)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.renderSlow(BatchedEntryRendererManager.java:257)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.render(BatchedEntryRendererManager.java:144)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.render(BatchedEntryRendererManager.java:118)
	at me.shedaniel.rei.impl.client.gui.widget.region.EntryStacksRegionWidget.method_25394(EntryStacksRegionWidget.java:135)
	at me.shedaniel.rei.impl.client.gui.widget.favorites.FavoritesListWidget.method_25394(FavoritesListWidget.java:214)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.renderWidgets(ScreenOverlayImpl.java:347)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.method_25394(ScreenOverlayImpl.java:244)
	at me.shedaniel.rei.impl.client.registry.screen.ScreenRegistryImpl$5.render(ScreenRegistryImpl.java:375)
	at me.shedaniel.rei.impl.client.registry.screen.DefaultScreenOverlayRenderer.lambda$onApplied$5(DefaultScreenOverlayRenderer.java:91)
	at me.shedaniel.rei.impl.client.registry.screen.DefaultScreenOverlayRenderer.lambda$new$1(DefaultScreenOverlayRenderer.java:61)
	at net.minecraft.class_465.handler$zkf000$architectury$renderBackground(class_465.java:1219)
	at net.minecraft.class_465.method_25394(class_465.java:100)
	at net.minecraft.class_485.method_25394(class_485.java:27)
	at net.minecraft.class_490.method_25394(class_490.java:88)
	at net.minecraft.class_437.method_47413(class_437.java:110)
	at net.minecraft.class_757.method_3192(class_757.java:945)
	at net.minecraft.class_310.method_1523(class_310.java:1219)
	at net.minecraft.class_310.method_1514(class_310.java:802)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
```

great mod hope this helps thanks bye!!

**Comment by Erdragh** [comment]
This is a bug in Create‘s REI implementation. I’ve PR-ed a fix into the 1.18 branch of Create to fix this for Create: Astral. ~~I’ll try and port the fix to the other versions supported by Create when I get the chance.~~

For reference: the fix PR: https://github.com/Fabricators-of-Create/Create/pull/1490

Edit: I won’t have to port it forwards myself, Create will do that once they put out the next patch.

**Comment by thrilliams** [comment]
thanks! i ended up finding this bug annoying enough to patch myself before doing the smart thing and checking to see if it was replicated anywhere else, but i appreciate the response and am glad that this is in the process of being fixed!
